RecoLocalMuon/GEMSegment/plugins/GEMSegAlgo.h

```diff
@@ -0,0 +1,86 @@
+#ifndef GEMRecHit_GEMSegAlgo_h
+#define GEMRecHit_GEMSegAlgo_h
+
+/**
+ * \class GEMSegAlgo
+ *
+ * This algorithm is very basic no attemp to deal with ambiguities , noise etc.
+ * The GEM track segments (actually more correct would be: GEM correlated hits)
+ * is built out of the rechits in two GEM layers (GE1/1) or three GEM layers (GE2/1)
+ * as the GEM Ensabmle .<BR>
+ *
+ *  \authors Piet Verwilligen 
+ *  updated by Jason Lee to use general segment fitter, MuonSegFit
+ */
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <RecoLocalMuon/GEMSegment/plugins/GEMSegmentAlgorithm.h>
+#include <DataFormats/GEMRecHit/interface/GEMRecHit.h>
+#include "FWCore/Utilities/interface/Exception.h"
+#include <deque>
+#include <vector>
+
+class MuonSegFit;
+
+class GEMSegAlgo : public GEMSegmentAlgorithm {
```

Comment:
this should be more creative and more descriptive in the naming (if anything, GEMSegAlgo could be a typedef to GEMSegmentAlgorithm, not a different implementation)